lib/extractStyles.js

```diff
@@ -145,74 +146,23 @@ function extractStyles(src, staticNamespace, getClassNameAndComment) {
       classNameAndComment.comment ? '/* ' + classNameAndComment.comment + ' */\n  ' : ''
     );
 
-    var hoverStyles = {};
-    var focusStyles = {};
-    var activeStyles = {};
-
-    // Check entry for any props with pseudo selector prefixes, move them to
-    // respective pseudo style object
-    Object.keys(entry.staticAttributes).forEach(function(key) {
-      if (key.indexOf('hover') === 0) {
-        // Strip the key of the pseudo selector prefix and transform first char
-        // to lowercase to ensure correct markup generation and autoprefixing
-        var prop = key.substr(5);
-        hoverStyles[prop.charAt(0).toLowerCase() + prop.slice(1)] = entry.staticAttributes[key];
-        // Remove the key from main style entry
-        delete entry.staticAttributes[key];
-      }
-      if (key.indexOf('focus') === 0) {
-        var prop = key.substr(5);
-        focusStyles[prop.charAt(0).toLowerCase() + prop.slice(1)] = entry.staticAttributes[key];
-        delete entry.staticAttributes[key];
-      }
-      if (key.indexOf('active') === 0) {
-        var prop = key.substr(6);
-        activeStyles[prop.charAt(0).toLowerCase() + prop.slice(1)] = entry.staticAttributes[key];
-        delete entry.staticAttributes[key];
+    var explodedStyles = explodePseudoStyles(entry.staticAttributes);
+
+    for (var key in explodedStyles) {
+      if (explodedStyles.hasOwnProperty(key)) {
+        if (explodedStyles[key]) {
+          // Only print ':<pseudo>' when necessary
+          var pseudoSelector = key.startsWith('base') ? '' : ':' + key;
+          css += (
+            '.' + className + ' {\n  ' +
+              comment +
+              CSSPropertyOperations.createMarkupForStyles(
+                autoprefix(explodedStyles[key])
+              ).split(';').join(';\n  ').trim() +
+              '\n}\n\n'
```

Comment:
@petehunt Should we also pull this into a helper function and change `explodePseudoStyles()` into just `createCSS(styles, prefix, comment) -> (css)` ?